src/lrf.rs

```diff
@@ -444,46 +432,31 @@ pub struct RestorationState {
 
 impl RestorationState {
   pub fn new(fi: &FrameInvariants, input: &Frame) -> Self {
-    // unlike the other loop filters that operate over the padded
-    // frame dimensions, restoration filtering and source pixel
-    // accesses are clipped to the original frame dimensions
-    let mut clipped_cfg:[PlaneConfig; 3] = [input.planes[0].cfg.clone(),
-                                    input.planes[1].cfg.clone(),
-                                    input.planes[2].cfg.clone()];
-    clipped_cfg[0].width = fi.width;
-    clipped_cfg[0].height = fi.height;
-    
-    let PlaneConfig { xdec, ydec, .. } = clipped_cfg[1];
-
-    clipped_cfg[1].width = fi.width + (1 << xdec >> 1) >> xdec;
-    clipped_cfg[1].height = fi.height + (1 << ydec >> 1) >> ydec;
-    clipped_cfg[2].width = fi.width + (1 << xdec >> 1) >> xdec;
-    clipped_cfg[2].height = fi.height + (1 << ydec >> 1) >> ydec;
+    // Currrently opt for smallest possible restoration unit size (1 superblock)
 
-    // Currrently opt for smallest possible restoration unit size
-    let lrf_y_shift = if fi.sequence.use_128x128_superblock {1} else {2};
-    let lrf_uv_shift = lrf_y_shift + if xdec>0 && ydec>0 {1} else {0};
-    let lrf_type: [u8; PLANES] = [RESTORE_SWITCHABLE, RESTORE_SWITCHABLE, RESTORE_SWITCHABLE];
-    let unit_size: [usize; PLANES] = [RESTORATION_TILESIZE_MAX >> lrf_y_shift,
-                                      RESTORATION_TILESIZE_MAX >> lrf_uv_shift,
-                                      RESTORATION_TILESIZE_MAX >> lrf_uv_shift];
+    // if 128x128 superblock is enabled, we need a shift because currently
+    // SuperBlockOffset is always defined in terms of 64x64 superblocks
+    let sb_log2 = if fi.sequence.use_128x128_superblock {1} else {0};
```

Comment:
@xiphmont, Do you think this is correct?
( (1 << sb_log2) gives # of SBs in a restoration unit, both horizontally and vertically)

Reply:
So, if use_128x128_superblock is true, restoration unit size = 256x256, and it has 2 SBs in hor and ver?

Reply:
> So, if use_128x128_superblock is true, restoration unit size = 256x256, and it has 2 SBs in hor and ver?

Currently, the restoration unit size is always exactly 1 superblock:
 - 128×128 if `use_128x128_superblock`;
 - 64×64 otherwise.

`(1 << sb_log2)` gives the size of the restoration unit always expressed in "superblocks" of size 64×64, even if `use_128x128_superblock` (the caller always gives superblock offsets expressed in [64x64 blocks](https://github.com/xiph/rav1e/blob/07032e6b9702f05446629422750337a49794499f/src/context.rs#L1129-L1153)).

Concretely, if `use_128x128_superblock`, then `sb_log2 = 1`. `1 << sb_log2 == 2`, so the restoration unit is a square with side length 2 (a square 2×2 "superblocks-64x64"). So the restoration unit is 128×128.

Otherwise, `sb_log2 = 0`. `1 << sb_log2 == 1`, so the restoration unit is a square with side length 1 (a single "superblock-64x64"). So the restoration unit is 64×64.

(I kept the existing behavior.)

Reply:
I think "always expressed in "superblocks" of size 64×64, even if use_128x128_superblock" confused me, and can do possibly some others as well.
Adding '64' somewhere with 'sb_log2', such as 'sb64_log2', would help readers.

In addition, the comment for sb_log2 member,  "// (1 << sb_log2) gives the number of superblocks (having size (1 << SUPERBLOCK_TO_PLANE_SHIFT))" will needs update once SUPERBLOCK_TO_PLANE_SHIFT became 7.

Reply:
I think I am still correct.
If it says : "// (1 << sb_log2) gives the number of **64x64** superblocks (having size (1 << **6**))" ... ", then I agree it is correct.

However, with the current comment, as you commented above like "if use_128x128_superblock, then sb_log2 = 1. 1 << sb_log2 == 2", which is 2 of 64 pels, i.e. 1 << 6. While "(1 << SUPERBLOCK_TO_PLANE_SHIFT) is 128 if use_128x128_superblock is true.

Reply:
> If it says : "// (1 << sb_log2) gives the number of **64x64** superblocks (having size (1 << **6**))" ... ", then I agree it is correct.

I think this would become incorrect then, because if `SUPERBLOCK_TO_PLANE_SHIFT` becomes `7`, then `SuperBlockOffset` will be expressed in terms of superblocks 128×128, and `(1 << sb_log2)` will gives the number of superblocks 128×128.

However, this one would need to be updated:

    // SuperBlockOffset is always defined in terms of 64x64 superblocks

(but it would make the comment less clear)

Reply:
> Currently, the restoration unit size is always exactly 1 superblock:

And that's temporary only as a bootstrap to get the code working for now.  It's a mode we will almost never use in practice once LRF is complete.

Getting all this sizing logic correct was a minor pain; stripping it out is a mistake.

Reply:
> as you commented above like "if use_128x128_superblock, then sb_log2 = 1. 1 << sb_log2 == 2", which is 2 of 64 pels, i.e. 1 << 6. While "(1 << SUPERBLOCK_TO_PLANE_SHIFT) is 128 if use_128x128_superblock is true.

Yes, the comment on this PR would need to be updated in that case (but not the comment in the code) :wink:

Reply:
>> Currently, the restoration unit size is always exactly 1 superblock:
> And that's temporary only as a bootstrap to get the code working for now. It's a mode we will almost never use in practice once LRF is complete.

The purpose of `sb_log2` is to encode the number of superblocks per restoration unit. Currently it is always 0 (or 1 due to the "superblock" VS "superblock-64x64" quirk), but the idea was to change it. Isn't it sufficient?